src/transformStateWithClones.js

```diff
@@ -7,7 +7,56 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+  let objectBox = {};
+  let objectBox2 = {};
+
+  for (const key in state) {
+    objectBox2[key] = state[key];
+  }
+
+  for (let i = 0; i < actions.length; i++) {
+    objectBox = {};
+
+    for (const key in objectBox2) {
+      objectBox[key] = objectBox2[key];
+    }
+
+    switch (actions[i].type) {
+      case 'addProperties':
+        for (const key in actions[i].extraData) {
+          objectBox[key] = actions[i].extraData[key];
+        }
+        resultArray.push(objectBox);
+        objectBox2 = objectBox;
+        break;
+
+      case 'removeProperties':
+        for (const key of actions[i].keysToRemove) {
+          if (!objectBox[key]) {
+            continue;
+          }
+
+          delete objectBox[key];
+        }
+        resultArray.push(objectBox);
+        objectBox2 = objectBox;
+        break;
```

Comment:
Correct all comments from previous PR here as well

Reply:
I started see more of mine frequent mistakes.